lib/happo/public/HappoDiffs.jsx

```diff
@@ -133,7 +133,7 @@ function SelectedView({ image, selectedView }) {
 }
 SelectedView.propTypes = {
   image: PropTypes.shape(imageShape).isRequired,
-  selectedView: PropTypes.string.isRequired,
+  selectedView: PropTypes.oneOf(Object.keys(VIEWS).map(key => VIEWS[key])).isRequired,
```

Comment:
Yeah that thought had crossed my mind. At least for the values map.

On Wed, Aug 31, 2016, 10:01 PM Henric Trotzig <notifications@github.com>
wrote:

> In lib/happo/public/HappoDiffs.jsx
> <https://github.com/Galooshi/happo/pull/125#discussion_r77114202>:
>
> > @@ -133,7 +133,7 @@ function SelectedView({ image, selectedView }) {
> >  }
> >  SelectedView.propTypes = {
> >    image: PropTypes.shape(imageShape).isRequired,
> > -  selectedView: PropTypes.string.isRequired,
> > +  selectedView: PropTypes.oneOf(Object.keys(VIEWS).map(key => VIEWS[key])).isRequired,
>
> we do this in two places now, so it might be worth initializing it at the
> top somewhere.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Galooshi/happo/pull/125/files/f357d4c39ad4cc412cd7a57f21f01fc8144638ab..a1ae6b813226730012ac82a8763fccf6f14df8d5#r77114202>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAL7zkcI6sGR_yWFdO1AmfKVW6PokuBCks5qllwxgaJpZM4Jxqz4>
> .
>